src/main/java/mate/academy/dao/jdbc/CarDaoJdbc.java

```diff
@@ -0,0 +1,185 @@
+            if (resultSet.next()) {
+                car = getCar(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get car by id: " + id, e);
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String getAllCar = "SELECT * FROM cars c "
+                + "JOIN manufacturers m ON c.manufacturer_id = m.manufacturer_id "
+                + "WHERE c.is_deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(getAllCar)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get list of car", e);
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateCars = "UPDATE cars SET car_model = ?, manufacturer_id = ? "
+                + "WHERE car_id = ? and is_deleted = FALSE;";
+        String deleteCarsDrivers = "DELETE FROM cars_drivers WHERE car_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(updateCars);
+                PreparedStatement statementDelete =
+                        connection.prepareStatement(deleteCarsDrivers)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+            statementDelete.setLong(1, car.getId());
+            statementDelete.executeUpdate();
+            createCarsDrivers(car);
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update table with this car: " + car, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteQuery = "UPDATE cars SET is_deleted = TRUE WHERE car_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(deleteQuery)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete car by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getCars = "SELECT * FROM cars_drivers cd "
+                + "JOIN cars c ON c.car_id = cd.car_id "
+                + "JOIN manufacturers m ON c.manufacturer_id = m.manufacturer_id "
+                + "WHERE cd.driver_id = ? AND c.is_deleted = FALSE;";
```

Comment:
let's join `drivers` table as well - we shouldn't fetch info if the driver is deleted